operators/src/plot/temporal_raster_mean_plot.rs

```diff
@@ -0,0 +1,438 @@
+                found: self.raster_sources.len()
+            }
+        );
+
+        Ok(InitializedTemporalRasterMeanPlot {
+            params: self.params,
+            result_descriptor: PlotResultDescriptor {},
+            raster_sources: self
+                .raster_sources
+                .into_iter()
+                .map(|o| o.initialize(context))
+                .collect::<Result<Vec<_>>>()?,
+            vector_sources: vec![],
+            state: (),
+        }
+        .boxed())
+    }
+}
+
+/// The initialization of `TemporalRasterMeanPlot`
+pub type InitializedTemporalRasterMeanPlot =
+    InitializedOperatorImpl<TemporalRasterMeanPlotParams, PlotResultDescriptor, ()>;
+
+impl InitializedOperator<PlotResultDescriptor, TypedPlotQueryProcessor>
+    for InitializedTemporalRasterMeanPlot
+{
+    fn query_processor(&self) -> Result<TypedPlotQueryProcessor> {
+        let input_processor = self.raster_sources[0].query_processor()?;
+        let time_position = self.params.time_position;
+        let measurement = self.raster_sources[0]
+            .result_descriptor()
+            .measurement
+            .clone();
+
+        let processor = call_on_generic_raster_processor!(input_processor, raster => {
+            TemporalRasterMeanPlotQueryProcessor { raster, time_position, measurement }.boxed()
+        });
+
+        Ok(TypedPlotQueryProcessor::JsonVega(processor))
+    }
+}
+
+/// A query processor that calculates the `TemporalRasterMeanPlot` about its input.
+pub struct TemporalRasterMeanPlotQueryProcessor<P: Pixel> {
+    raster: Box<dyn RasterQueryProcessor<RasterType = P>>,
+    time_position: TemporalRasterMeanPlotTimePosition,
+    measurement: Measurement,
+}
+
+#[async_trait]
+impl<P: Pixel> PlotQueryProcessor for TemporalRasterMeanPlotQueryProcessor<P> {
+    type OutputFormat = PlotData;
+
+    fn plot_type(&self) -> &'static str {
+        TEMPORAL_RASTER_MEAN_PLOT_NAME
+    }
+
+    async fn plot_query<'a>(
+        &'a self,
+        query: QueryRectangle,
+        ctx: &'a dyn QueryContext,
+    ) -> Result<Self::OutputFormat> {
+        let means =
+            Self::calculate_means(self.raster.query(query, ctx), self.time_position).await?;
+
+        let plot = Self::generate_plot(means, self.measurement.clone())?;
+
+        let plot_data = plot.to_vega_embeddable(false)?;
+
+        Ok(plot_data)
+    }
+}
+
+impl<P: Pixel> TemporalRasterMeanPlotQueryProcessor<P> {
+    async fn calculate_means(
+        mut tile_stream: BoxStream<'_, Result<RasterTile2D<P>>>,
+        position: TemporalRasterMeanPlotTimePosition,
+    ) -> Result<BTreeMap<TimeInstance, MeanCalculator>> {
+        let mut means: BTreeMap<TimeInstance, MeanCalculator> = BTreeMap::new();
+
+        while let Some(tile) = tile_stream.next().await {
+            let tile = tile?;
+
+            let time = Self::time_interval_projection(tile.time, position);
+
+            let mean = means.entry(time).or_default();
```

Comment:
maybe initialize the Calculators once and then just expect them here. Might save millions of checks whether it's there. Or maybe the complier is smart enough.

Reply:
I have to do this at least for every time step and I don't know them in advance. So I can't pull it out of the loop. I have to check it at least once for every tile and this is done here.